tests/tasks/test_juniperalarmtask.py

```diff
@@ -0,0 +1,140 @@
+        address="127.0.0.1",
+        community="public",
+        port=snmp_test_port,
+    )
+    task = JuniperAlarmTask(device=device)
+    device_state = state.devices.get(device_name=device.name)
+    device_state.enterprise_id = 11
+
+    await task.run()
+
+    assert state.devices[device.name].alarms is None
+
+
+@patch("zino.state.devices", DeviceStates())
+@pytest.mark.asyncio
+async def test_juniper_alarm_saves_alarm_count_in_device_state(snmpsim, snmp_test_port):
+    from zino import state
+
+    device = PollDevice(
+        name="buick.lab.example.org",
+        address="127.0.0.1",
+        community="juniper-alarm",
+        port=snmp_test_port,
+    )
+    task = JuniperAlarmTask(device=device)
+    device_state = state.devices.get(device_name=device.name)
+    device_state.enterprise_id = 2636
+
+    await task.run()
+
+    assert state.devices[device.name].alarms["yellow"] == 0
+    assert state.devices[device.name].alarms["red"] == 2
+
+
+@patch("zino.state.devices", DeviceStates())
+@pytest.mark.asyncio
+async def test_juniper_alarm_overrides_alarm_count_in_device_state(snmpsim, snmp_test_port):
+    from zino import state
+
+    device = PollDevice(
+        name="buick.lab.example.org",
+        address="127.0.0.1",
+        community="juniper-alarm",
+        port=snmp_test_port,
+    )
+    task = JuniperAlarmTask(device=device)
+    device_state = state.devices.get(device_name=device.name)
+    device_state.enterprise_id = 2636
+    device_state.alarms = {
+        "yellow": "1",
+        "red": "3",
+    }
+
+    await task.run()
+
+    assert state.devices[device.name].alarms["yellow"] == 0
+    assert state.devices[device.name].alarms["red"] == 2
+
+
+@patch("zino.state.devices", DeviceStates())
+@patch("zino.state.events", Events())
+@pytest.mark.asyncio
+async def test_juniper_alarm_creates_alarm_events(snmpsim, snmp_test_port):
+    from zino import state
+
+    device = PollDevice(
+        name="buick.lab.example.org",
+        address="127.0.0.1",
+        community="juniper-alarm",
+        port=snmp_test_port,
+    )
+    task = JuniperAlarmTask(device=device)
+    device_state = state.devices.get(device_name=device.name)
+    device_state.enterprise_id = 2636
+
+    await task.run()
+
+    yellow_event = state.events.get(device_name=device.name, port="yellow", event_class=AlarmEvent)
+    red_event = state.events.get(device_name=device.name, port="red", event_class=AlarmEvent)
+
+    assert yellow_event
+    assert red_event
+    assert yellow_event.alarm_count == 0
+    assert red_event.alarm_count == 2
+
+
+@patch("zino.state.devices", DeviceStates())
+@patch("zino.state.events", Events())
+@pytest.mark.asyncio
+async def test_juniper_alarm_updates_alarm_events(snmpsim, snmp_test_port):
+    from zino import state
+
+    device = PollDevice(
+        name="buick.lab.example.org",
+        address="127.0.0.1",
+        community="juniper-alarm",
+        port=snmp_test_port,
+    )
```

Comment:
this could be defined as a fixture so you dont have to define the same thing in every test